### PR TITLE
[Merged by Bors] - golf: replace some `apply foo.mpr` by `rw [foo]`

### PR DESCRIPTION
**Pull request by grunweg**
Sometimes, that line can be golfed into the next line.
Inspired by a [comment](https://github.com/leanprover-community/mathlib4/pull/11208#discussion_r1530467335) of @loefflerd; any decisions are my own.

-------

(I'm happy to drop some of these changes if they're actually dubious.) 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
!bench

File: Mathlib/Control/Functor/Multivariate.lean [hunk 1]
```diff
@@ -128,7 +128,7 @@ theorem exists_iff_exists_of_mono {P : F α → Prop} {q : F β → Prop}
   · refine ⟨f <$$> u, ?_⟩
     apply (h₁ u).mp h₂
   · refine ⟨g <$$> u, ?_⟩
-    apply (h₁ _).mpr _
+    rw [h₁ _]
```
Comment (fpvandoorn) [inline Mathlib/Control/Functor/Multivariate.lean:131]:
```suggestion
    rw [h₁]
```
if it compiles

File: Mathlib/Data/Set/Intervals/Disjoint.lean [hunk 2]
```diff
@@ -214,16 +214,15 @@ theorem IsGLB.biUnion_Ici_eq_Ioi (a_glb : IsGLB s a) (a_not_mem : a ∉ s) :
...
     exact ⟨y, hys, hyx.le⟩
 #align is_glb.bUnion_Ici_eq_Ioi IsGLB.biUnion_Ici_eq_Ioi
 
 theorem IsGLB.biUnion_Ici_eq_Ici (a_glb : IsGLB s a) (a_mem : a ∈ s) :
     ⋃ x ∈ s, Ici x = Ici a := by
   refine' (iUnion₂_subset fun x hx => _).antisymm fun x hx => _
   · exact Ici_subset_Ici.mpr (mem_lowerBounds.mp a_glb.1 x hx)
-  · apply mem_iUnion₂.mpr
-    exact ⟨a, a_mem, hx⟩
+  · apply mem_iUnion₂.mpr ⟨a, a_mem, hx⟩
```
Comment (fpvandoorn) [inline Mathlib/Data/Set/Intervals/Disjoint.lean:225]:
```suggestion
  · exact mem_iUnion₂.mpr ⟨a, a_mem, hx⟩
```

**Review by fpvandoorn — APPROVED** [review]
I think all of these are an improvement.
I have two minor suggestions (if they compile).

bors d+

**Comment by grunweg** [comment]
Thank you for the fast review. Both suggestions compile.
bors r+

**Pull request closed without merging**